tests/test_process.py

```diff
@@ -221,6 +221,37 @@ def test_process_list_widget(multiply_add_completed_workchain):
     ProcessListWidget()
 
 
+@pytest.mark.usefixtures("aiida_profile_clean")
+def test_process_monitor(
+    multiply_add_process_builder_ready, daemon_client, await_for_process_completeness
+):
+    """Test ProcessMonitor with a simple `WorkChainNode`"""
+    from aiidalab_widgets_base.process import ProcessMonitor
+
+    ProcessMonitor()
+
+    # Stopping the daemon and submitting the process.
+    daemon_client.stop_daemon(wait=True)
+    process = engine.submit(multiply_add_process_builder_ready)
+
+    test_variable = False
+
+    def f():
+        nonlocal test_variable
+        test_variable = True
+
+    widget = ProcessMonitor(value=process.uuid, callbacks=[f])
+
+    # Starting the daemon and waiting for the process to complete.
+    daemon_client.start_daemon()
+    await_for_process_completeness(process)
+
+    widget.join()  # Make sure the thread is finished.
+
+    assert test_variable
+    daemon_client.stop_daemon(wait=True)
```

Comment:
Seems this will close the daemon for the test session and lead to the failure of other tests that use the daemon. Maybe for every tests that use the daemon using `started_daemon_client` fixture? As documented in https://aiida.readthedocs.io/projects/aiida-core/en/latest/topics/plugins.html#plugin-test-fixtures

Reply:
I need the `daemon_client` fixture as I stop and start it a few times during the test. I updated the PR.

Reply:
`daemon_client` has it all, so I just used it. 🤷